### PR TITLE
EventGrid: New preview API version with support for IPFiltering in Domains & Topics.

### PR DESCRIPTION
**Pull request by grandhiramesh**
EventGrid: New preview API version with support for IPFiltering in Domains & Topics.

oTopicProperties: Following new properties are introduced:
allowTrafficFromAllIPs
inboundIpRules
oDomainProperties: Following new properties are introduced:
allowTrafficFromAllIPs
inboundIpRules
oMinor change in the description fields:
oInboundIpRule: Following new model with the following properties are introduced:
IpMask
Action

### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [x] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [x] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

**Comment by grandhiramesh** [comment]
@kalyanaj & @akning-ms 

the semantic validation is throwing error on an existing property and it is no impact on the proposed changes in the swagger as part of the feature release.

Also, the mentioned property is not a mandatory field and by design not a required property.

This looks like an OAV tool issue. Can you please bypass the error

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json [hunk 1]
```diff
@@ -0,0 +1,3798 @@
...
+          },
+          {
+            "$ref": "#/parameters/TopParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "schema": {
+              "$ref": "#/definitions/TopicsListResult
```
Comment (ryansbenson) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json]: looks like you removed 'readonly': 'true'.

Was this intentional?

Comment (ryansbenson) [inline specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json]: Are their other values? If not, does this need to be exposed?

**Review by ryansbenson — COMMENTED** [review]
A few questions, everything else looks good.

**Review by zikalino — APPROVED** [review]
compared with previous 2020-01-01-preview.
minimal changes.
lgtm

File: specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json (hunk 1 above)
Comment (grandhiramesh) [reply specification/eventgrid/resource-manager/Microsoft.EventGrid/preview/2020-04-01-preview/EventGrid.json]: this is intentional. May expand few more during GA version  of this feature.

**Pull request merged**